src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
```

Comment:
Ignore my comment about the possible optimization for now. But get to know what "lazy initialization/evaluation" means ;)